lib/AI-Agent.xml

```diff
@@ -5,9 +5,16 @@
         <BuiltIn enabled="true">
             <HTTP enabled="true"/>
         </BuiltIn>
-        <Class name="uk.gov.hmcts.auth.provider.service.api.auth.AuthService">
-            <Method name="lease" reportCaughtExceptions="true" reportExecutionTime="true"/>
-            <Method name="verify" reportCaughtExceptions="true" reportExecutionTime="true"/>
+        <Class name="uk.gov.hmcts.auth.provider.service.api.auth.jwt.JwtRS256Tool">
+            <Method name="issueTokenForSubject" />
+            <Method name="verifyAndExtractSubject" />
+        </Class>
+        <Class name="uk.gov.hmcts.auth.provider.service.api.auth.jwt.JwtHS512Tool">
+            <Method name="issueTokenForSubject" />
+            <Method name="verifyAndExtractSubject" />
+        </Class>
+        <Class name="uk.gov.hmcts.auth.provider.service.api.auth.totp.TotpAuthenticator">
+            <Method name="isOneTimePasswordValid" />
```

Comment:
how can we be sure all this stays in sync with actual class/method/package names?

Reply:
do we actually care about tracking execution times of those methods?
anyone?

Reply:
I would care. Being able to point out separate areas where (for some reason) code is not working as expected. Previous service methods were not that useful as they are first and only level entries in API call chain which are already tracked by default

Reply:
I know that we can't use wildcards for method names so this is the way we will have use this atm. 
How do we know this is working as I can see jar was missing previously.It would be great if you could list down steps/screenshot may be which shows all the methods in the config are getting tracked?

Reply:
Regarding names I agree - shame we cannot name them properly so the map visibility would look nice.
Apparently I was properly drooling using wrong app boot around java agent hence no updated data in app insights. After noticing that, here the result:

<img width="164" alt="application-map" src="https://user-images.githubusercontent.com/1420191/39436072-e832e174-4c94-11e8-97f7-1fd2cfa5b861.png">